### ISSUE TITLE
Why is progress.png so noisy ? 

### ISSUE THREAD

**Issue by SimJeg** [MIC-DKFZ/nnUNet#1216]
Hello,

Disclaimer : I am very new to the nnU-Net framework.

I noticed that the learning curves are very noisy, in my experiments or in the issues I saw on the repository. Looking at the code, the default is to get 50 random batches from the validation. From 1 epoch to the other, the validation set is different, and so the metrics, making the curve noisy.

Is there a simple way to get the same 50 batches for every epoch ? From my understanding it is not possible as the data loader (DataLoader3D) randomly picks samples from the the whole dataset (np.random.choice).

In my experience, segmentation is less prone to overfitting than other tasks and you can train the network forever so I am not sure how important the issue is.

**Comment by SimJeg** [comment]
I tried to get the same validation batches between to runs by : 

- Setting a random seed for numpy (for DataLoader3D to select the same keys between two runs) 
- Setting the parameter seeds_val for the MultiThreadedAugmenter (as in data_augmentation_noDA.py)

I checked it works by looking if the validation loss is the same when setting deterministic=True. It worked if I use no samples for training (num_batches_per_epoch=0). It seems the training is not deterministic even if it was supposed to be so I am not 100% sure this works because I couldn't reproduce the validation loss with num_batches_per_epoch > 0

**Comment by Karol-G** [comment]
Hey @SimJeg,

I do not recommend to always use the same batches for validation as that can cause overfitting. 

If you want to compare two nnU-Net models against each other then I recommend fully training them both and to check the final Dice scores in the summary.json of each fold located under nnunet_experiments/DatasetXXX/trainerXXX/fold_XXX/validation/summary.json. The metrics reported during training are only an estimate and are not suited for comparison.

Best,
Karol
